### ISSUE TITLE
Halyard with custom sizing is not working as expected

### ISSUE THREAD

**Issue by rverma-nikiai** [spinnaker/spinnaker#4778]
### Issue Summary:
As per doc when creating a hal config we can specify either container name or service name to specify resource limits. But this option is missing using hal command.

### Additional Details:

---

**_Please delete the instructions below this line prior to submitting_**

Instructions:

* An issue is not a place to ask questions. Please use [Slack](http://join.spinnaker.io) or [Stack Overflow](http://stackoverflow.com/questions/tagged/spinnaker). 
* Before you open an issue, please [check if a similar issue already exists](https://github.com/spinnaker/spinnaker/issues) or has been closed before.
* Make sure you have read through the [Spinnaker FAQ](https://www.spinnaker.io/community/faqs/) and [Halyard FAQ](https://www.spinnaker.io/setup/quickstart/faq/) to provide as much information as possible.

Descriptions:

* Issue Summary: A brief description of what you're seeing.
* Cloud Provider: AWS, GCP, Kubernetes, Azure, Cloud Foundry, etc. Please assign a label from the right so your issue can be properly sorted.
* Environment: As much information about your Spinnaker environment and configuration that might be relevant to the issue. For example: "I am running Spinnaker using the Amazon images to deploy into AWS and GCP."
* Feature Area: Notifications, Pipelines, UI, Jenkins, etc. Please assign a label from the right so your issue can be properly sorted.
* Description: The behavior you expect to see, and the actual behavior.
* Steps to reproduce: Ideally, an isolated way to reproduce the behavior (example: GitHub repository with code isolated to the issue that anyone can clone to observe the problem). If not possible, as much information as possible to see this behavior.
* Additional Details: Additional information such as screenshots and exception logs.

**Comment by rverma-nikiai** [comment]
Also it includes a service called monitoring daemon, if we specify any resource settings there it simply removes any resource allocation.

**Comment by costimuraru** [comment]
> and there are no options to set it only for cloud-driver

Do you want to set the limits only for the clouddriver **container** and not necessarily for the entire pod? Support was added recently for this.

**Comment by rverma-nikiai** [comment]
Yes, I really want to add resources for clouddriver container and not to monitoring container.

**Comment by costimuraru** [comment]
@rverma-nikiai check out https://www.spinnaker.io/reference/halyard/commands/#hal-config-deploy-component-sizing-clouddriver-edit. @mtweten added support for this a few weeks ago via https://github.com/spinnaker/halyard/pull/1387

You can now pass `--container-limits-cpu`, `--container-limits-memory`, `--container-requests-cpu` ...

**Comment by ofen** [comment]
@rverma-nikiai, hi

Is there any way to set resource limits to sidecar monitoring-daemon only?
